### PR TITLE
Fixing issue for Archive if, multiple backends in SSP

### LINKED ISSUE #1284: [v1.3.2 Jerba RC2 Testing] Unable to archive an object to a bucket with Service Plan with both AWS and Azure backends attached and Azure backend is selected..
**Issue/Feature Description:**
When a service plan is created with both AWS and Azure backends attached and bucket is created on the Azure backend, while archiving an object, the storage class sent is incorrect. Instead of `Archive` the storage class `GLACIER` is sent to the cloud backend which fails.

**Why this issue to fixed / feature is needed(give scenarios or use cases):**
The user will not be able to archive an object .
**How to reproduce, in case of a bug:**

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**
archive-aws_azure_tier

### PR DESCRIPTION
**Pull request by PravinRanjan10**
This PR for fixing a issue when SSP is enabled:
* When there are multiple backends in a SSP and buckets are created on different backends of SSP.
* Internal logic, was selecting the random backenId for bucket. But it should pick only those backend in which buckets are created.
* This PR will select only that backends in which buckets are crated and based on backend Type, storageClass will be filled for archiving.

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind new feature
> /kind bug fix
> /kind cleanup
> /kind revert change
> /kind design
> /kind documentation
> /kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #1284 

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
> /kind TESTED
> /kind NOT-TESTED

**Test Report**:

**Special notes for your reviewer**:

### CODE REVIEW

File: api/pkg/s3/objectput.go [hunk 1]
```diff
@@ -56,18 +56,22 @@ func (s *APIService) ObjectPut(request *restful.Request, response *restful.Respo
 
 	if isArchive == "Archive" && bucketMeta.Tiers != "" {
 		adminCtx := common.GetAdminContext()
-		backendId := s.GetBackendIdFromTier(adminCtx, bucketMeta.Tiers)
-		backendMeta, err := s.backendClient.GetBackend(adminCtx, &backend.GetBackendRequest{Id: backendId})
-		if err != nil {
-			log.Error("the selected backends from tier doesn't exists.")
-			response.WriteError(http.StatusInternalServerError, err)
+		backendMeta, backendErr := s.backendClient.ListBackend(adminCtx, &backend.ListBackendRequest{
+			Offset: 0,
```
Comment (kumarashit) [inline api/pkg/s3/objectput.go:60]: This is not used

File: api/pkg/s3/objectput.go [hunk 2]
```diff
@@ -56,18 +56,22 @@ func (s *APIService) ObjectPut(request *restful.Request, response *restful.Respo
 
 	if isArchive == "Archive" && bucketMeta.Tiers != "" {
 		adminCtx := common.GetAdminContext()
-		backendId := s.GetBackendIdFromTier(adminCtx, bucketMeta.Tiers)
-		backendMeta, err := s.backendClient.GetBackend(adminCtx, &backend.GetBackendRequest{Id: backendId})
-		if err != nil {
-			log.Error("the selected backends from tier doesn't exists.")
-			response.WriteError(http.StatusInternalServerError, err)
+		backendMeta, backendErr := s.backendClient.ListBackend(adminCtx, &backend.ListBackendRequest{
```
Comment (kumarashit) [inline api/pkg/s3/objectput.go:59]: Where do we ensure that the backend has bucket?

Comment (kumarashit) [inline api/pkg/s3/objectput.go:59]: Where do we ensure that the backend has bucket?

Comment (PravinRanjan10) [reply api/pkg/s3/objectput.go:59]: we are passing in filter, bucketMeta.defaultLocation

Comment (kumarashit) [reply api/pkg/s3/objectput.go:59]: got it

**Review by kumarashit — APPROVED** [review]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM

**Pull request merged**